### PR TITLE
Automate GitHub releases

### LINKED ISSUE #2231: releases section not being updated
image

cc @katspaugh this should be fixed with npm publish action because it creates new tags right?

### PR DESCRIPTION
**Pull request by katspaugh**
@thijstriemstra thanks for the docs link, it worked nicely 👍 

I've also reorganised the steps a bit, so that each command has a separate title.

Here's the build that published release 4.4.0: https://github.com/katspaugh/wavesurfer.js/runs/2271665459?check_suite_focus=true

Closes #2231.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -23,14 +23,35 @@ jobs:
...
+          fi
+      - name: Tag
+        run: |
+          if [ -f version.txt ]; then
+            git tag $(cat version.txt)
             git push --tags
+          fi
+      - name: Publish to NPM
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
```
Comment (thijstriemstra) [inline .github/workflows/publish.yml:41]: can you add some spaces around this var (e.g. see line 48)

**Review by thijstriemstra — CHANGES_REQUESTED** [review]
Nice! The release name should be corrected though, see https://github.com/katspaugh/wavesurfer.js/releases/tag/4.4.0

**Comment by katspaugh** [comment]
Fixed the name format ✅

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -23,14 +23,36 @@ jobs:
...
-          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
+      - name: Extract version
         run: |
           OLD_VERSION=$(npm show wavesurfer.js version)
           NEW_VERSION=$(node -p 'require("./package.json").version')
           if [ $NEW_VERSION != $OLD_VERSION ]; then
-            git tag "$NEW_VERSION"
+            echo $NEW_VERSION > version.txt
+          fi
+      - name: Tag
```
Comment (thijstriemstra) [inline .github/workflows/publish.yml:33]: CMake it more descriptive, e.g. `Create Git tag`

**Comment by thijstriemstra** [comment]
Maybe this is a bonus-points kind of thing, but for some reason the author name is `github-actions`, would be nice if it's the person tagging or creating the release. Renaming `github-actions` to something more general should be fine as well, e.g. `wavesurfer.js`

Screenshot from 2021-04-05 23-25-00

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -23,14 +23,36 @@ jobs:
...
+        run: |
+          if [ -f version.txt ]; then
+            git tag $(cat version.txt)
```
Comment (thijstriemstra) [inline .github/workflows/publish.yml:36]: adding some echo statements describing the flow you added would help as well

**Review by thijstriemstra — APPROVED** [review]
thanks!

**Pull request merged**